sdk/schemaregistry/Microsoft.Azure.Data.SchemaRegistry.JsonSchema/src/SchemaRegistryJsonSerializer.cs

```diff
@@ -0,0 +1,688 @@
+        ///   An error occurred while attempting to communicate with the Schema Registry service.
+        /// </exception>
+        /// <exception cref="Exception">
+        ///   The data did not adhere to the JSON schema, or the schema itself was invalid.
+        ///   The <see cref="Exception.InnerException"/> will contain the underlying exception from the <see cref="SchemaRegistryJsonSchemaGenerator"/> class.
+        /// </exception>
+        public async ValueTask<MessageContent> SerializeAsync(
+            object data,
+            Type dataType = default,
+            Type messageType = default,
+            CancellationToken cancellationToken = default)
+            => await SerializeInternalAsync(data, null, dataType, messageType, true, cancellationToken).ConfigureAwait(false);
+
+        /// <summary>
+        /// Serializes the message data as JSON and stores it in <see cref="MessageContent.Data"/>. The <see cref="MessageContent.ContentType"/>
+        /// will be set to "application/json+schemaId" where schemaId is the ID of the schema.
+        /// </summary>
+        /// <remarks>
+        /// This method doesn't generate the schema from <typeparamref name="TData"/>, it instead uses the schema retrieved using the
+        /// <see cref="SchemaRegistryClient"/>. It is recommended to implement <see cref="SchemaRegistryJsonSchemaGenerator.ValidateAgainstSchema(object, Type, string)"/>
+        /// to validate <paramref name="data"/> adheres to its intended schema.
+        /// </remarks>
+        /// <typeparam name="TMessage">The <see cref="MessageContent"/> type to serialize the data into.</typeparam>
+        /// <typeparam name="TData">The type of the data to serialize.</typeparam>
+        /// <param name="data">The data to serialize to JSON and serialize into the message.</param>
+        /// <param name="schemaId">The ID of the schema to use when serializing the data.</param>
+        /// <param name="cancellationToken">An optional <see cref="CancellationToken"/> instance to signal the request to cancel the operation.</param>
+        /// <exception cref="InvalidOperationException">
+        ///   This can occur if the <c>groupName</c> was not specified when constructing the <see cref="SchemaRegistryJsonSerializer"/>.
+        ///   It can also occur if the <typeparamref name="TMessage"/> does not have a public parameterless constructor.
+        /// </exception>
+        /// <exception cref="RequestFailedException">
+        ///   An error occurred while attempting to communicate with the Schema Registry service.
+        /// </exception>
+        /// <exception cref="Exception">
+        ///   The data did not adhere to the JSON schema, or the schema itself was invalid.
+        ///   The <see cref="Exception.InnerException"/> will contain the underlying exception from the <see cref="SchemaRegistryJsonSchemaGenerator"/> class.
+        /// </exception>
+        public TMessage SerializeWithSchemaId<TMessage, TData>(
```

Comment:
Thinking maybe we should ditch these Id methods for now, and consider adding them later if people want them?

Reply:
+1 to that idea